willofsteel/async_client.py

```diff
@@ -0,0 +1,333 @@
+    async def get_player_inventory(self) -> dict[ItemType, int]:
+        """
+        Retrieve player inventory.
+
+        """
+        data, status = await self.request("GET", "/inventory", self.headers)
+        if status == 200:
+            logging.debug(
+                f"Got player inventory data successfully: {data}. Returning with converting to Model.")
+            return {convert_str_to_IT(item_id): amount for item_id, amount in data["items"].items()}
+        else:
+            parse_error(data["detail"])
+            print(data["detail"])
+            print("This error was not automatically detected, please report this to the maintainers (or fix it yourself)!")
+
+    async def get_player_army(self) -> dict[UnitType, int]:
+        """
+        Retrieve player army.
+
+        """
+        data, status = await self.request("GET", "/army", self.headers)
+        if status == 200:
+            logging.debug(
+                f"Got player army data successfully: {data}. Returning with converting to Model.")
+            return {convert_str_to_UT(unit_type): amount for unit_type, amount in data["units"].items()}
+        else:
+            parse_error(data["detail"])
+            print(data["detail"])
+            print("This error was not automatically detected, please report this to the maintainers (or fix it yourself)!")
+
+    async def get_outposts(self) -> list[Outpost]:
+        """
+        Retrieve all outposts.
+
+        """
+        data, status = await self.request("GET", "/outpost/list", self.headers)
+        if status == 200:
+            logging.debug(
+                f"Got outposts data successfully: {data}. Returning with converting to Model.")
+            return [Outpost.from_data(outpost) for outpost in data["outposts"]]
+        else:
+            parse_error(data["detail"])
+            print(data["detail"])
+            print("This error was not automatically detected, please report this to the maintainers (or fix it yourself)!")
+
+    async def get_alliance(self) -> dict:
+        """
+        Retrieve alliance information.
+
+        Returns
+        -------
+        Optional[:class:`~willofsteel.types.Alliance`]
+
+        """
+        data, status = await self.request("GET", "/alliance", headers=self.headers)
+        if status == 400:
+            raise NotInAlliance
+        logging.debug(
+            f"Got alliance data successfully: {data}. Returning with converting to Model.")
+        return Alliance.from_response(data)
+
+    async def update_alliance_name(self, new_name: str) -> bool:
+        """
+        Update the name of the alliance.
+
+        Parameters
+        ----------
+        new_name: :class:`str`
+            The new name of the alliance.
+
+        Returns
+        -------
+        :class:`bool`
+
+        """
+        if len(new_name) > 32:  # this is not an official limit. just a wrapper limit for now
+            raise LimitExceeded(32, "name")
+        headers = self.headers
```

Comment:
```suggestion
        headers = self.headers.copy()
```

Create a copy of headers, not edit them.